packages/pwa-kit-dev/src/utils/logger.js

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Array defining the order of log levels from lowest to highest severity.
+ * The order of log levels is important. It determines the logging threshold.
+ * @type {string[]}
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+
+// Default log level for the logging threshold. It can be any of the LOG_LEVELS values.
+const DEFAULT_LOG_LEVEL = 'info'
+
+// Default log format to use if not explicitly provided. It can be 'JSON' or 'TEXT'.
+const DEFAULT_LOG_FORMAT = 'JSON'
+
+const isServerSide = typeof window === 'undefined'
+
+// Configuration object defining logging settings for server/client environments
+const loggerConfig = {
+    server: {
+        logLevel: isServerSide
+            ? process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL
+            : DEFAULT_LOG_LEVEL,
+        format: isServerSide
+            ? process.env.PWAKIT_LOG_FORMAT || DEFAULT_LOG_FORMAT
+            : DEFAULT_LOG_FORMAT
+    },
+    client: {
+        logLevel: DEFAULT_LOG_LEVEL,
+        format: DEFAULT_LOG_FORMAT
+    }
+}
```

Comment:
Is there ever an occasion where you need to know the "logLevel" for both client and server at the same time? If the answer is no, then you should probably do something like this.

```
const loggerConfig = isServerSide ? 
    {
        logLevel: process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL,
        format: process.env.PWAKIT_LOG_FORMAT || DEFAULT_LOG_FORMAT
    } : {
        logLevel: DEFAULT_LOG_LEVEL,
        format: DEFAULT_LOG_FORMAT
    }
}
```

Reply:
Thanks! makes sense, it's a good simplification. I added the suggestion.